addons/delivery/tests/test_delivery_cost.py

```diff
@@ -452,6 +453,7 @@ def test_base_on_rule_currency_is_converted(self):
         })
 
         self.env['res.currency.rate'].with_company(nook_inc).create({
+            'name': datetime.date(2000, 1, 1),
```

Comment:
I do not see how it can fix the issue with date, maybe it is better to use a decorator
`@freeze_time` for this test to ensure the creation time?

Reply:
@vchu-odoo 
Sorry I guess the PR could have used more explanation. 
It works because in `res.currency` the `name` field doubles as the date and we filter for rates that are before today when we convert rates. I think what happens is that after 22:00 in utc we are tomorrow in Brussels so the currency is not taken into account if we just created it and didn't specify the `name`/date. (default date is now)
https://github.com/odoo/odoo/blob/d8a20717c830a7cddaad4889bec182f0fa5f786a/odoo/addons/base/models/res_currency.py#L124-L132

Not sure if you would prefer for me to amend my commit with a better explanation or go for a `freeze_time` based solution as you suggested.

Reply:
Thanks for the clear explanation :+1: 

To avoid having to find this explanation again to give it to future readers of this test, let's go with a `freeze_time` that is so extensively used to address this kind of issue that it's self-documenting at this point.